### ISSUE TITLE
2.0.0-M.3 c-xml-client

### ISSUE THREAD

**Issue by anton-johansson** [stoicflame/enunciate#128]
If I DON'T have ```<c-xml-client disabled="true" />```  in my ```enunciate.xml```, I get a nasty exception when running ```mvn enunciate:docs```

```
[INFO] Scanning for projects...
[INFO] 
[INFO] Using the builder org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder with a thread count of 1
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Viskan Distanshandel System AB :: API - Application Interface - CXF Connector 0.0.0.Dev-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- enunciate-maven-plugin:2.0.0-M.3:docs (default-cli) @ <artifactId> ---
[INFO] [ENUNCIATE] Using enunciate configuration at C:\Users\<username>\Projects\<project>\enunciate.xml
[INFO] Reflections took 1066 ms to scan 38 urls, producing 658 keys and 658 values 
sep 05, 2015 11:12:26 EM freemarker.log._JULLoggerFactory$JULLogger error
SEVERE: Error executing FreeMarker template
FreeMarker template error:
The following has evaluated to null or missing:
==> choice.baseType.qname  [in template "jar:file:/C:/Users/<username>/.m2/repository/com/webcohesion/enunciate/enunciate-c-xml-client/2.0.0-M.3/enunciate-c-xml-client-2.0.0-M.3.jar!/com/webcohesion/enunciate/modules/c_client/client-complex-type.fmt" at line 525, column 177]

----
Tip: It's the step after the last dot that caused this error, not those before it.
----
Tip: If the failing expression is known to be legally refer to something that's null or missing, either specify a default value like myOptionalVar!myDefault, or use [#if myOptionalVar??]when-present[#else]when-missing[/#if]. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
----

----
FTL stack trace ("~" means nesting-related):
	- Failed at: ${choice.baseType.qname.namespaceURI}  [in template "jar:file:/C:/Users/<username>/.m2/repository/com/webcohesion/enunciate/enunciate-c-xml-client/2.0.0-M.3/enunciate-c-xml-client-2.0.0-M.3.jar!/com/webcohesion/enunciate/modules/c_client/client-complex-type.fmt" in macro "writeComplexTypeImpl" at line 525, column 175]
	- Reached through: @libComplexType.writeComplexTypeImpl ...  [in template "jar:file:/C:/Users/<username>/.m2/repository/com/webcohesion/enunciate/enunciate-c-xml-client/2.0.0-M.3/enunciate-c-xml-client-2.0.0-M.3.jar!/com/webcohesion/enunciate/modules/c_client/api.fmt" at line 48, column 13]
	- Reached through: @file name=cFileName charset="utf-8"  [in template "jar:file:/C:/Users/<username>/.m2/repository/com/webcohesion/enunciate/enunciate-c-xml-client/2.0.0-M.3/enunciate-c-xml-client-2.0.0-M.3.jar!/com/webcohesion/enunciate/modules/c_client/api.fmt" at line 14, column 1]
----

Java stack trace (for programmers):
----
freemarker.core.InvalidReferenceException: [... Exception message was already printed; see it above ...]
	at freemarker.core.InvalidReferenceException.getInstance(InvalidReferenceException.java:116)
	at freemarker.core.UnexpectedTypeException.newDesciptionBuilder(UnexpectedTypeException.java:60)
	at freemarker.core.UnexpectedTypeException.<init>(UnexpectedTypeException.java:40)
	at freemarker.core.NonHashException.<init>(NonHashException.java:46)
	at freemarker.core.Dot._eval(Dot.java:45)
	at freemarker.core.Expression.eval(Expression.java:78)
	at freemarker.core.Expression.evalAndCoerceToString(Expression.java:82)
	at freemarker.core.DollarVariable.accept(DollarVariable.java:40)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.MixedContent.accept(MixedContent.java:62)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
	at freemarker.core.IfBlock.accept(IfBlock.java:48)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.MixedContent.accept(MixedContent.java:62)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
	at freemarker.core.IfBlock.accept(IfBlock.java:48)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.MixedContent.accept(MixedContent.java:62)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
	at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
	at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
	at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.MixedContent.accept(MixedContent.java:62)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
	at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
	at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
	at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.MixedContent.accept(MixedContent.java:62)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
	at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.MixedContent.accept(MixedContent.java:62)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.Macro$Context.runMacro(Macro.java:178)
	at freemarker.core.Environment.invoke(Environment.java:700)
	at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
	at freemarker.core.IfBlock.accept(IfBlock.java:48)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
	at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
	at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
	at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
	at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
	at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
	at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
	at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
	at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
	at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.MixedContent.accept(MixedContent.java:62)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.Environment$NestedElementTemplateDirectiveBody.render(Environment.java:2255)
	at com.webcohesion.enunciate.util.freemarker.FileDirective.execute(FileDirective.java:59)
	at freemarker.core.Environment.visit(Environment.java:377)
	at freemarker.core.UnifiedCall.accept(UnifiedCall.java:104)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.MixedContent.accept(MixedContent.java:62)
	at freemarker.core.Environment.visit(Environment.java:312)
	at freemarker.core.Environment.process(Environment.java:290)
	at freemarker.template.Template.process(Template.java:312)
	at com.webcohesion.enunciate.modules.c_client.CXMLClientModule.processTemplate(CXMLClientModule.java:237)
	at com.webcohesion.enunciate.modules.c_client.CXMLClientModule.call(CXMLClientModule.java:174)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.subjects.SubjectSubscriptionManager$SubjectObserver.onCompleted(SubjectSubscriptionManager.java:232)
	at rx.subjects.ReplaySubject.onCompleted(ReplaySubject.java:404)
	at rx.Observable$35.onCompleted(Observable.java:7458)
	at rx.observers.SafeSubscriber.onCompleted(SafeSubscriber.java:81)
	at rx.internal.operators.OperatorDoOnEach$1.onCompleted(OperatorDoOnEach.java:53)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:110)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.subjects.SubjectSubscriptionManager$SubjectObserver.onCompleted(SubjectSubscriptionManager.java:232)
	at rx.subjects.ReplaySubject.onCompleted(ReplaySubject.java:404)
	at rx.Observable$35.onCompleted(Observable.java:7458)
	at rx.observers.SafeSubscriber.onCompleted(SafeSubscriber.java:81)
	at rx.internal.operators.OperatorDoOnEach$1.onCompleted(OperatorDoOnEach.java:53)
	at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
	at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
	at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 10.052 s
[INFO] Finished at: 2015-09-05T23:12:26+01:00
[INFO] Final Memory: 43M/454M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.3:docs (default-cli) on project <artifactId>: Execution default-cli of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.3:docs failed: com.webcohesion.enunciate.EnunciateException: freemarker.core.InvalidReferenceException: The following has evaluated to null or missing:
[ERROR] ==> choice.baseType.qname  [in template "jar:file:/C:/Users/<username>/.m2/repository/com/webcohesion/enunciate/enunciate-c-xml-client/2.0.0-M.3/enunciate-c-xml-client-2.0.0-M.3.jar!/com/webcohesion/enunciate/modules/c_client/client-complex-type.fmt" at line 525, column 177]
[ERROR] 
[ERROR] ----
[ERROR] Tip: It's the step after the last dot that caused this error, not those before it.
[ERROR] ----
[ERROR] Tip: If the failing expression is known to be legally refer to something that's null or missing, either specify a default value like myOptionalVar!myDefault, or use [#if myOptionalVar??]when-present[#else]when-missing[/#if]. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
[ERROR] ----
[ERROR] 
[ERROR] ----
[ERROR] FTL stack trace ("~" means nesting-related):
[ERROR] - Failed at: ${choice.baseType.qname.namespaceURI}  [in template "jar:file:/C:/Users/<username>/.m2/repository/com/webcohesion/enunciate/enunciate-c-xml-client/2.0.0-M.3/enunciate-c-xml-client-2.0.0-M.3.jar!/com/webcohesion/enunciate/modules/c_client/client-complex-type.fmt" in macro "writeComplexTypeImpl" at line 525, column 175]
[ERROR] - Reached through: @libComplexType.writeComplexTypeImpl ...  [in template "jar:file:/C:/Users/<username>/.m2/repository/com/webcohesion/enunciate/enunciate-c-xml-client/2.0.0-M.3/enunciate-c-xml-client-2.0.0-M.3.jar!/com/webcohesion/enunciate/modules/c_client/api.fmt" at line 48, column 13]
[ERROR] - Reached through: @file name=cFileName charset="utf-8"  [in template "jar:file:/C:/Users/<username>/.m2/repository/com/webcohesion/enunciate/enunciate-c-xml-client/2.0.0-M.3/enunciate-c-xml-client-2.0.0-M.3.jar!/com/webcohesion/enunciate/modules/c_client/api.fmt" at line 14, column 1]
[ERROR] ----
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
Exception in thread "pool-1-thread-1" java.lang.NoClassDefFoundError: com/webcohesion/enunciate/modules/idl/JaxbSchemaFile
	at com.webcohesion.enunciate.modules.idl.IDLModule.call(IDLModule.java:169)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.subjects.SubjectSubscriptionManager$SubjectObserver.onCompleted(SubjectSubscriptionManager.java:232)
	at rx.subjects.ReplaySubject.onCompleted(ReplaySubject.java:404)
	at rx.Observable$35.onCompleted(Observable.java:7458)
	at rx.observers.SafeSubscriber.onCompleted(SafeSubscriber.java:81)
	at rx.internal.operators.OperatorDoOnEach$1.onCompleted(OperatorDoOnEach.java:53)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:110)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
	at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
	at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)
Caused by: java.lang.ClassNotFoundException: com.webcohesion.enunciate.modules.idl.JaxbSchemaFile
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:259)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:235)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:227)
	... 33 more
```

**Comment by anton-johansson** [comment]
Very similar to #129

**Comment by anton-johansson** [comment]
Might be a duplicate of #118

**Comment by stoicflame** [comment]
Fixed at 6d21563.

**Comment by stoicflame** [comment]
Fix was included in [Milestone 4](https://github.com/stoicflame/enunciate/releases/tag/v2.0.0-M.4). The c client should be automatically skipped when incompatible conditions are detected. Thanks for the report, and please report any new issues.
